src/server/terrain/structures/Lake.zig

```diff
@@ -0,0 +1,88 @@
+const std = @import("std");
```

Comment:
This lake has a bunch of bugs:
Caves can be seen through the bottom of a lake:
![Screenshot at 2024-06-03 11-31-56](https://github.com/PixelGuys/Cubyz/assets/43880493/c310b7a0-703c-488f-a5ea-a730f0a89bda)
The lake interacts badly with other structures:
![Screenshot at 2024-06-03 11-55-34](https://github.com/PixelGuys/Cubyz/assets/43880493/c7f98265-d418-4445-9f4e-07d05a4a8e22)
![Screenshot at 2024-06-03 11-55-49](https://github.com/PixelGuys/Cubyz/assets/43880493/2f27949a-8d05-4862-9322-f38b4e904ced)
![Screenshot at 2024-06-03 11-56-00](https://github.com/PixelGuys/Cubyz/assets/43880493/ff2df8ce-e3c9-4967-830a-e9afe640469a)
The lake itself sometimes cuts off at a chunk border (note that this is not #203, since it breaks close to the center of the lake):

![Screenshot at 2024-06-03 11-57-04](https://github.com/PixelGuys/Cubyz/assets/43880493/694b39df-632c-4730-b42a-75d4b8faf62e)


I think it might be better to do structures like this during map generation. This would require some more work though, like introducing a new structure system for MapGen structures.